### PR TITLE
added department permissions

### PR DESCRIPTION
**Pull request by VadimZharikov**
- On get/{id} action adds the permission parameter

### CODE REVIEW

File: HRMS/HRMS.WebApi/Controllers/EmployeesController.cs [hunk 1]
```diff
@@ -24,10 +25,35 @@ public EmployeesController(IEmployeeService employeeService, IMapper mapper)
...
         // GET: api/<EmployeesController>
         [HttpGet("{id}")]
-        public async Task<EmployeeViewModel> GetEmployee(int id)
+        public async Task<string> GetEmployee(int id)
         {
-            var employee = await employeeService.GetEmployee(id);
-            return _mapper.Map<Employee, EmployeeViewModel>(employee);
+            var employee = _mapper.Map <Employee, EmployeeViewModel>(await employeeService.GetEmployee(id));
+            var result = JObject.FromObject(employee);
+            switch (employee.DepartmentId)
```
Comment (nantonov) [inline HRMS/HRMS.WebApi/Controllers/EmployeesController.cs:32]: All business logic must be in the service

File: HRMS/HRMS.WebApi/Controllers/EmployeesController.cs [hunk 2]
```diff
@@ -24,10 +25,35 @@ public EmployeesController(IEmployeeService employeeService, IMapper mapper)
...
+            {
+                case 1:
+                    {
+                        ;
+                        result.Add("Permissions", "Read, Write, Delete");
```
Comment (nantonov) [inline HRMS/HRMS.WebApi/Controllers/EmployeesController.cs:37]: Please, use Enum instead of Strings

**Review by nantonov — APPROVED** [review]

**Review by nantonov — CHANGES_REQUESTED** [review]
🔵

**Comment by VadimZharikov** [comment]
- Excluded logic from controller

File: HRMS/HRMS.BLL/Entities/Employee.cs [hunk 3]
```diff
@@ -10,5 +12,6 @@ public class Employee
         public string Position { get; set; }
         public string Phone { get; set; }
         public int DepartmentId { get; set; }
+        public IEnumerable<string> Permissions { get; set; }
```
Comment (nantonov) [inline HRMS/HRMS.BLL/Entities/Employee.cs:15]: Better to use IEnumerable of Enum with the same values

**Review by nantonov — CHANGES_REQUESTED** [review]

**Comment by VadimZharikov** [comment]
Changed permissions property to Permissions enum type

**Review by nantonov — APPROVED** [review]

**Pull request merged**